### DISCUSSION TITLE
Custom mail with record data when creating new record.

### DISCUSSION THREAD

**Discussion by Akasiek** [pocketbase/pocketbase#2005] [Q&A] [answered]
Hi guys!
I'm trying to make this custom mails when new record is created in collection. I created the function and all but I'm trying to also insert new record information. For example if new record is created with column called "name" it should send mail with some like this:
"New records was created. Name values is: foo".

Are there any templates that can do it? Maybe insert variable data into HTML? I know that inside pocketbase you can use `{TOKEN}` and others in mail templates.

Sorry for maybe a stupid question but I don't have any experience in Go.

For now I have something like this

```go
// main.go
package main

import (
    "log"
    "net/mail"

    "github.com/pocketbase/pocketbase"
    "github.com/pocketbase/pocketbase/core"
    "github.com/pocketbase/pocketbase/tools/mailer"
)

func main() {
    app := pocketbase.New()

    app.OnRecordAfterCreateRequest("neogage_contact").Add(func(e *core.RecordCreateEvent) error {
        messageHtml := "<div>
        <h1>Nowa wiadomość kontaktowa</h1>
        </div>"

        message := &mailer.Message{
            From: mail.Address{
                Address: app.Settings().Meta.SenderAddress,
                Name:    app.Settings().Meta.SenderName,
            },
            To:       []mail.Address{{Address: "office@neogage.com", Name: "Neogage Office"}},
            Subject:  "Nowa wiadomość kontaktowa NEOGAGE.com",
            HTML:     messageHtml,
        }

        return app.NewMailClient().Send(message)
    })


    if err := app.Start(); err != nil {
        log.Fatal(err)
    }
}
```

**Comment by ganigeorgiev** [comment, marked as answer]
> Are there any templates that can do it? I know that inside pocketbase you can use {TOKEN} and others in mail templates.

PocketBase doesn't come with its own template engine and uses the Go std library.

It is in my todo and I may consider exporting the internal `mails.resolveTemplateContent()` helper, but for now you
can check [`text/template`](https://pkg.go.dev/text/template) (_or its wrapper [`html/template`](https://pkg.go.dev/html/template)_) or use any other external library like [`valyala/quicktemplate`](https://github.com/valyala/quicktemplate), [`gobuffalo/plush`](https://github.com/gobuffalo/plush), etc.

Here is an example using the std `text/template`:

```go
params := struct {
  Name string
}{
  Name: "John Doe",
}

var content bytes.Buffer

t, err := template.New("mail").Parse(`Hello {{.Name}}!`)
if err != nil {
  return err
}

if err := t.Execute(&content, params); err != nil {
  return err
}

log.Println(content.String()) // Hello John Doe
```

**Reply by Akasiek** [reply]
Thanks for the replay. I'm still looking how to get info about new record. Can I maybe do it with this `e` variable inside `app.OnRecordAfterCreateRequest("neogage_contact").Add(func(e *core.RecordCreateEvent)`?

**Reply by ganigeorgiev** [reply]
Yes, you can access the created record fields using `e.Record`. For example, if your collection has a `name` field, you can access it like `e.Record.GetString("name")`.
You can find a list with other record helpers in https://pocketbase.io/docs/record-methods/#getset-record-fields.

**Reply by Akasiek** [reply]
Ok. I think I can do it from here. Thank you so much for help!
